rotkehlchen/chain/evm/decoding/compound/v3/decoder.py

```diff
@@ -209,10 +248,69 @@ def _decode_repay_or_borrow_event(self, context: DecoderContext) -> DecodingOutp
                 f'{context.transaction.tx_hash.hex()}.',
             )
 
+        action_items = []  # also create an action item for the spend of the cTokens
+        if paired_event is not None and action_from_event_type is not None:
+            action_items.append(ActionItem(
+                action='transform',
+                from_event_type=action_from_event_type,
+                from_event_subtype=HistoryEventSubType.NONE,
+                asset=compound_token,
+                to_event_subtype=action_to_event_subtype,
+                to_notes=action_to_notes,
+                to_counterparty=CPT_COMPOUND_V3,
+                paired_event_data=(paired_event, False),
+            ))
+
+        return DecodingOutput(action_items=action_items)
+
+    def _decode_collateral_movemement(self, context: DecoderContext) -> DecodingOutput:
+        """Decode a compound v3 supply/withdraw collateral events"""
+        for event in context.decoded_events:
+            if (
+                event.event_type in {HistoryEventType.SPEND, HistoryEventType.RECEIVE} and
+                event.event_subtype == HistoryEventSubType.NONE
+            ):
+                transfer_event = event
+                event.counterparty = CPT_COMPOUND_V3
+                if event.event_type == HistoryEventType.SPEND:
+                    notes_action = 'Enable'
+                    event.event_type = HistoryEventType.DEPOSIT
+                    event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                    event.notes = f'Deposit {event.balance.amount} {event.asset.symbol_or_name()} into compound v3'  # noqa: E501
+                else:
+                    notes_action = 'Disable'
+                    event.event_type = HistoryEventType.WITHDRAWAL
+                    event.event_subtype = HistoryEventSubType.REMOVE_ASSET
+                    event.notes = f'Withdraw {event.balance.amount} {event.asset.symbol_or_name()} from compound v3'  # noqa: E501
+
+                collateral_event = self.base.make_event_next_index(
+                    tx_hash=context.transaction.tx_hash,
+                    timestamp=context.transaction.timestamp,
+                    event_type=HistoryEventType.INFORMATIONAL,
+                    event_subtype=HistoryEventSubType.NONE,
+                    balance=Balance(),
+                    asset=event.asset,
+                    location_label=event.location_label,
+                    counterparty=event.counterparty,
+                    notes=f'{notes_action} {event.asset.symbol_or_name()} as collateral on compound v3',  # noqa: E501
+                    address=event.address,
+                )
+                break
+        else:
+            log.error(
+                f'Could not find any compound v3 supply/withdraw collateral event in tx '
+                f'{context.transaction.tx_hash.hex()}.',
+            )
+
+        context.decoded_events.append(collateral_event)
```

Comment:
I might be missing something but I believe there is a scenario in the code if something goes bad where collateral_event is udnefined here

Reply:
right, adding `return DEFAULT_DECODING_OUTPUT` now

Reply:
Here this also seems to not have changed. TGhe collateral_event could be missing here where it is used